packages/devp2p/src/dpt/dpt.ts

```diff
@@ -148,6 +157,20 @@ export class DPT {
     }
   }
 
+  /**
+   * Add peer to a confirmed list of peers (peers meeting some
+   * level of quality, e.g. being on the same network) to allow
+   * for a more selective findNeighbours request and sending
+   * (with activated `onlyConfirmed` setting)
+   *
+   * @param id Unprefixed hex id
+   */
+  confirmPeer(id: string) {
+    if (this._confirmedPeers.size < 5000) {
+      this._confirmedPeers.add(id)
```

Comment:
Peers are never removed from `_confirmedPeers`, so at some point we cannot confirm new peers since the set is full? Should the peer id be removed on `removePeer`?

Reply:
Yeah, that was the lazy version assuming a very slow growth. But removing on `removePeer` definitely make sense, will add. 👍